src/main/java/me/dinowernli/grpc/polyglot/io/MessageWriter.java

```diff
@@ -65,7 +75,14 @@ public void onError(Throwable t) {
   @Override
   public void onNext(T message) {
     try {
-      output.write(jsonPrinter.print(message) + MESSAGE_SEPARATOR);
+      String outputString = jsonPrinter.print(message);
+      // Only separate messages if there are more than one
```

Comment:
I don't actually think this is the right thing to do. To make things easier to parse, I believe we alway want to append a separator, even if there is nothing else coming. Can discuss in person.